contracts/src/bridge/Outbox.sol

```diff
@@ -135,22 +138,51 @@ contract Outbox is DelegateCallAware, IOutbox {
         uint256 l2Timestamp,
         uint256 value,
         bytes calldata data
-    ) external virtual {
-        bytes32 outputId;
-        {
-            bytes32 userTx = calculateItemHash(
-                l2Sender,
-                to,
-                l2Block,
-                l1Block,
-                l2Timestamp,
-                value,
-                data
-            );
-
-            outputId = recordOutputAsSpent(proof, index, userTx);
-            emit OutBoxTransactionExecuted(to, l2Sender, 0, index);
-        }
+    ) external {
+        bytes32 userTx = calculateItemHash(
+            l2Sender,
+            to,
+            l2Block,
+            l1Block,
+            l2Timestamp,
+            value,
+            data
+        );
+
+        recordOutputAsSpent(proof, index, userTx);
+
+        executeTransactionImpl(index, l2Sender, to, l2Block, l1Block, l2Timestamp, value, data);
+    }
+
+    /// @dev function used to simulate the result of a particular function call from the outbox
+    /// it is useful for things such as gas estimates.
```

Comment:
Maybe include a comment about how this function should include everything except proof validation - and the reason why it can't include proof validation